scripts/entities_extended_types_aliases/build_extended_aliases.ts

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env tsx
+import { writeFile } from 'node:fs/promises'
+import { difference, intersection, uniq, values, omit, cloneDeep } from 'lodash-es'
+import { prefixifyWd, getWdEntityUriNumericId } from '#controllers/entities/lib/prefix'
+import { makeCachedSparqlRequest } from '#data/wikidata/make_sparql_request'
+import { absolutePath } from '#lib/absolute_path'
+import { cache_ } from '#lib/cache'
+import { newError } from '#lib/error/error'
+import { oneDay } from '#lib/time'
+import { getHashCode, objectEntries } from '#lib/utils/base'
+import { info, logError } from '#lib/utils/logs'
+import { primaryTypesAliases, type PluralizedEntityType } from '#lib/wikidata/aliases'
+import { extendedAliasesQueries, getExtendedAliasesQueriesHash } from '#scripts/entities_extended_types_aliases/extended_type_aliases_queries'
+import type { WdEntityId, WdEntityUri } from '#server/types/entity'
+
+const extendedTypesAliases = cloneDeep(primaryTypesAliases)
+const stats = {}
+
+// Force a cache refresh by prefixing this script call with `export INV_REFRESH_ENTITIES_TYPE_EXTENDED_ALIASES=true;`
+const refresh = process.env.INV_REFRESH_ENTITIES_TYPE_EXTENDED_ALIASES === 'true'
+
+for (const [ type, sparql ] of objectEntries(extendedAliasesQueries)) {
+  let typeExtendedAliases = await getTypeExtendedAliases(type, sparql)
+  // Ideally, the following adjustments should be done within the queries
+  // but not at the cost of risking to time out
+  if (type === 'series') {
+    typeExtendedAliases = difference(typeExtendedAliases, extendedTypesAliases.collections)
+  } else if (type === 'works') {
+    typeExtendedAliases = difference(typeExtendedAliases, [ ...extendedTypesAliases.series, ...extendedTypesAliases.collections ])
+  }
+  if (type === 'movements' || type === 'genres') {
+    typeExtendedAliases = dropOverlaps(type, typeExtendedAliases)
+  } else {
+    checkOverlaps(type, typeExtendedAliases)
+  }
+  extendedTypesAliases[type] = typeExtendedAliases.sort(byNumericId)
```

Comment:
What happens if something wrong happen (ie. a timeout) and `typeExtendedAliases` is actually empty (as it seems it could happen if `filteredIds` is not populated) ?
I cant a good suggestion to make sure `extended_types_aliases.json` is populated well, but hopefully you do?

Reply:
It's part of the motivation to track `extended_types_aliases.json`: if for some reason the build fails (because query.wikidata.org is down, or what not), it makes it still possible to use the last successful build until the build process can be fixed